test/unit/spec/VerifyCustomFactor_spec.js

```diff
@@ -0,0 +1,268 @@
+    expect(test.afterErrorHandler).toHaveBeenCalledTimes(1);
+    expect(test.afterErrorHandler.mock.calls[0]).toEqual([
+      {
+        controller: controller,
+      },
+      {
+        name: 'AuthApiError',
+        message: message,
+        statusCode: code,
+        xhr: resError,
+      },
+    ]);
+  }
+
+  function expectSubtitleToBe(test, desiredSubtitle) {
+    expect(test.form.subtitleText()).toBe(desiredSubtitle);
+  }
+
+  async function setup(settings, lastFailedChallengeFactorData, languagesResponse , startRouter) {
+    const setNextResponse = Util.mockAjax();
+    const baseUrl = 'https://foo.com';
+    const authClient = getAuthClient({
+      authParams: {
+        issuer: baseUrl,
+        transformErrorXHR: LoginUtil.transformErrorXHR
+      }
+    });
+    const successSpy = jest.fn();
+    const afterErrorHandler =  jest.fn();
+    const router = createRouter(baseUrl, authClient, successSpy, settings, startRouter);
+
+    router.on('afterError', afterErrorHandler);
+    if (startRouter) {
+      if (typeof startRouter !== 'boolean') {
+        throw new Error('startRouter parameter should be boolean');
+      }
+      await Expect.waitForPrimaryAuth();
+    }
+
+    setNextResponse(resAllFactors);
+    if (languagesResponse) {
+      setNextResponse(languagesResponse);
+    }
+    router.refreshAuthState('dummy-token');
+    await Expect.waitForMfaVerify();
+
+    router.appState.set('lastFailedChallengeFactorData', lastFailedChallengeFactorData);
+
+    setNextResponse(resChallengeClaimsProvider);
+    router.verifyClaimsFactor();
+    await Expect.waitForVerifyCustomFactor();
+
+    const $forms = $sandbox.find('.o-form');
+
+    let forms = _.map($forms, function(form) {
+      return new VerifyCustomFactorForm($(form));
+    });
+
+    if (forms.length === 1) {
+      forms = forms[0];
+    }
+    const beacon = new Beacon($sandbox);
+
+    return {
+      router: router,
+      form: forms,
+      beacon: beacon,
+      ac: authClient,
+      setNextResponse: setNextResponse,
+      successSpy: successSpy,
+      afterErrorHandler: afterErrorHandler,
+    };
+  }
+
+  Expect.describe('Claims Provider Factor', function() {
+    Expect.describe('when features.skipIdpFactorVerificationBtn is true', function() {
+      const lastFailedChallengeFactorData = {
+        errorMessage: 'Verify failed.'
+      };
+
+      let settings = {};
+
+      beforeEach(() => {
+        settings = { 'features.skipIdpFactorVerificationBtn': true };
+      });
+
+      itp('hides the button bar', function() {
+        return setup(settings)
+          .then(function(test) {
+            Expect.isNotVisible(test.form.buttonBar());
+          });
+      });
+
+      itp('shows the button bar when lastFailedChallengeFactorData is not null', function() {
+        return setup(settings ,lastFailedChallengeFactorData).then(function(test) {
+          expect(test.form.$('.o-form-button-bar').hasClass('hide')).toBe(false);
```

Comment:
nit: we can use `test.form.buttonBar()` instead of `test.form.$('.o-form-button-bar')`